src/main/java/org/candlepin/subscriptions/db/SubscriptionRepository.java

```diff
@@ -23,20 +23,30 @@
 import org.candlepin.subscriptions.db.model.Subscription;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 import java.util.List;
+import java.util.Optional;
 
 /**
  * Repository for Subscription Entities
  */
-public interface SubscriptionRepository extends JpaRepository<Subscription, String> {
+public interface SubscriptionRepository extends JpaRepository<Subscription,
+    Subscription.SubscriptionCompoundId> {
 
     /**
      * Object a set of subscriptions
      * @param ownerId the ownerId of the subscriptions
      * @param subscriptionIds the list of subscriptionIds to filter on
      * @return a list of subscriptions with the specified ownerId and a subscriptionId from the provided list
      */
-    List<Subscription> findByOwnerIdAndSubscriptionIdIn(String ownerId,
-        List<String> subscriptionIds);
+    @Query("SELECT s FROM Subscription s where s.endDate > CURRENT_TIMESTAMP " +
+        "AND s.ownerId = :ownerId AND s.subscriptionId IN :subscriptionIds")
+    List<Subscription> findActiveByOwnerIdAndSubscriptionIdIn(@Param("ownerId") String ownerId,
```

Comment:
Can a query method be used here instead of the hardcoded query?
Something like:
`findByEndDateAfterAndOwnerIdAndSubscriptionIdIn(Calendar.getInstance().getDate(), ownerId, subscriptionId)`

Reply:
Unsure of how this will work with the OffsetDateTime type

Reply:
OffsetDateTime's super-precision seems to be trimmed by the DB as well which makes precise comparison tricky too. That's why I took the JPQL current_timestamp there, but this does create the query annotation need which I definitely get wanting to avoid. @awood @kahowell preferences or thoughts on a clean way to handle this? The unit tests currently show this issue well as once something is persisted to the db, fractions of a millisecond are dropped as java's offsetdatetime is more precise than the db stores.

Reply:
Wonder if we can do something cheeky with extending OffsetDateTime or something and use it as a db converter, applying a different DateTimeFormatter.ofPattern without the milliseconds.

Or maybe there's a @Temporal annotation we can leverage.

If there isn't an obvious straightforward solution for this, my vote is to keep it the way you have it and we can mull over it later.  Probably not worth spinning wheels on it right now.

Reply:
Personally, I'm fine with what's written, FWIW, I think the precision loss wouldn't cause significant change in behvaior.